### PR TITLE
ENH: additional survey languages

### PR DESCRIPTION
**Pull request by wasade**
The script used to generate these is included and now in `utils/`

Important:

* `survey_question_response` was not localized. The column called `response` is implicitly "american". This was not renamed as it will have side effects, and to keep this PR contained, I felt it best to defer that.
* the language codes (e.g., "en_US") are not in place yet for the same reason, to keep this PR contained 

The columns added could not be made `not null` as there is an old survey which did not get translated. The present schema assumes each question is represented in all languages, which cannot be maintained given the legacy survey. In my opinion, this suggests a structural refactor may be a good idea to better support multi-language, however I think this is similarly out of scope right now to keep the PR maintained.

If there are no qualms with the statements on keeping the PR contained (...at a cool 19k line changes...), I'll open corresponding issues and link to this PR.

### CODE REVIEW

File: microsetta_private_api/db/patches/0052.sql [hunk 1]
```diff
@@ -0,0 +1,19329 @@
...
+UPDATE ag.survey_question_response 
+  SET spanish = 'pulgadas'
+  WHERE survey_question_id = 109 AND
+        display_index = 1;
+UPDATE ag.survey_question_response 
+  SET spanish = 'centímetros'
+  WHERE survey_question_id = 109 AND
+        display_index = 2;
+
+UPDATE ag.survey_question SET spanish = '¿Consume carne o productos lácteos  de animales tratados con antibióticos?'
```
Comment (dhakim87) [inline microsetta_private_api/db/patches/0052.sql]: Is there an extra space between "lácteos" and "de"

File: microsetta_private_api/util/format_language_patches.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+def _format_update_question(lang, qid, question):
+    return (f"UPDATE ag.survey_question SET {lang} = '{question}'\n"
+            f"  WHERE survey_question_id = {qid};\n\n")
+
+
+def _format_update_resp(lang, qid, response, response_index):
+    # french likes to use single quotes, and postgres wants ''
+    # to escape
+    response = response.replace("'", "''")
+    return (f"UPDATE ag.survey_question_response \n"
```
[the remark this answers is not in the document]
Comment (wasade) [reply microsetta_private_api/util/format_language_patches.py]: I agree, but in this case we're translating official documents. If you'd prefer, I can look at having psycopg2 generate the statements

File: microsetta_private_api/db/patches/0052.sql [hunk 3]
```diff
@@ -0,0 +1,19329 @@
...
+UPDATE ag.survey_question_response 
+  SET spanish = 'marzo'
+  WHERE survey_question_id = 111 AND
+        display_index = 3;
+UPDATE ag.survey_question_response 
+  SET spanish = 'abril'
+  WHERE survey_question_id = 111 AND
+        display_index = 4;
+UPDATE ag.survey_question_response 
+  SET spanish = 'Mayo'
```
[the remark this answers is not in the document]
Comment (wasade) [reply microsetta_private_api/db/patches/0052.sql]: Will fix

**Comment by wasade** [comment]
Thanks, @dhakim87! These are not Google translations, our IRB requires that they are official. I'd be happy to express this function directly with psycopg2 and have it under a new script called `migration_support.py`. However, given that the patch system at present only knowns how to execute `.sql` scripts, it's not obvious to me right now how to get this content represented in the database reliably short of writing a new patching system. What about instead we use psycopg2 to produce sanitized statements, and express that as a patch?

**Comment by wasade** [comment]
Sorry, actually, I'm not sure I understand the requested change. The double quote is a proper escape for postgres. Is there a different escape you had in mind?

File: microsetta_private_api/db/patches/0052.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhakim87) [reply microsetta_private_api/db/patches/0052.sql]: I guess it's just a coincidence that May -> Mayo in google translate and all other month names go to their lowercase versions?  ;)

[the remark this answers is not in the document]
Comment (wasade) [reply microsetta_private_api/db/patches/0052.sql]: Translation costs are per-word. To reduce costs, the country names, months and birth years were omitted. Yes, I did those specific items via google translate. The rest of the documents are translated through an official translation service and the materials were independently QC'd.

**Comment by dhakim87** [comment]
So it's a little strange, but psycopg2 doesn't always actually handle filling in all the arguments from what I can tell, it sends an escaped string and arguments to postgres, which then further handles escaping.  But you can get pretty close to the escaped strings with psycopg2's cursor.mogrify() (or possibly cursor.execute() followed by cursor.query()) and they'll probably be acceptable as raw sql.  

But honestly I think this is also an argument for adding the raw translation file and adding the translations in as a python migration function rather than a 19000 line set of inserts.  

There's also postgres's COPY/COPY TO command, that can directly build a table from a csv and vice versa, if you can convert the translations to an acceptable format, you can replace the whole change script with a one line command.

**Pull request merged**